### ISSUE TITLE
microk8s is not running. microk8s.inspect showing no error

### ISSUE THREAD

**Issue by chiehpower** [ubuntu/microk8s#2224]
My k8s can work on `channel=1.18` and `classic` version such as checking the status via the dashboard. 
However, it cannot work if I install the higher version such as v20 or v21. 

I encountered a similar issue with #886.

```
 microk8s.inspect
Inspecting Certificates
Inspecting services
  Service snap.microk8s.daemon-cluster-agent is running
  Service snap.microk8s.daemon-containerd is running
  Service snap.microk8s.daemon-apiserver is running
  Service snap.microk8s.daemon-apiserver-kicker is running
  Service snap.microk8s.daemon-control-plane-kicker is running
  Service snap.microk8s.daemon-proxy is running
  Service snap.microk8s.daemon-kubelet is running
  Service snap.microk8s.daemon-scheduler is running
  Service snap.microk8s.daemon-controller-manager is running
  Copy service arguments to the final report tarball
Inspecting AppArmor configuration
Gathering system information
  Copy processes list to the final report tarball
  Copy snap list to the final report tarball
  Copy VM name (or none) to the final report tarball
  Copy disk usage information to the final report tarball
  Copy memory usage information to the final report tarball
  Copy server uptime to the final report tarball
  Copy current linux distribution to the final report tarball
  Copy openSSL information to the final report tarball
  Copy network configuration to the final report tarball
Inspecting kubernetes cluster
  Inspect kubernetes cluster
Inspecting juju
  Inspect Juju
Inspecting kubeflow
  Inspect Kubeflow

Building the report tarball
  Report tarball is at /var/snap/microk8s/2143/inspection-report-20210503_160739.tar.gz
```

Here is the tar file.
[inspection-report-20210503_160739.tar.gz](https://github.com/ubuntu/microk8s/files/6413603/inspection-report-20210503_160739.tar.gz)

Thank you so much.

**Comment by balchua** [comment]
I see this in the logs

```console
03 16:07:32 10.1.2.110 microk8s.daemon-apiserver[905309]: I0503 16:07:32.278095  905309 log.go:181] [ERROR] dqlite: proxy: first: remote -> local: remote error: tls: bad certificate
```
@ktsakalozos apologies for the direct mention, do you have an idea what can be the cause of this?

**Comment by ktsakalozos** [comment]
The error reported is similar to the one we saw in [1]. 

When the cluster starts we need to issue certificates for dqlite, see [2] on how we do that. The first thing that stands out to me is the hostname of your machine. Do you think you could switch to a hostname with latin lowercase characters? Also a question, is there a proxy involved in your setup?


[1] https://github.com/ubuntu/microk8s/issues/1864#issue-777704513
[2] https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/actions/common/utils.sh#L549

**Comment by chiehpower** [comment]
Hi @balchua , 

Thanks for your reply! 
Yes, it is a single node cluster. I am trying to create a node and pods at the same device without using the cloud way.

Hi @ktsakalozos ,

Thanks for your suggestions!!
This is quite helpful!! because of some reasons that before I set the hostname as my internal IP, now it can work on newer versions of microk8s after I changed the hostname with Latin lowercase characters and reinstall again. 

Thanks a lot!
